File: 11-container-with-most-water/container-with-most-water.ts
function maxArea(height: number[]): number {

    let l:number=0;
    let r:number=height.length-1;

    let maxArea:number=0;

    while(l<r){

        maxArea= Math.max(maxArea,Math.min(height[l],height[r])*(r-l));

        if(height[l]<height[r]){
            l++;
        }
        else{
            r--;
        }
    }

    return maxArea;
    
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {

        int l=0;
        int r=height.length-1;

        int maxArea=0;

        while(l<r){

            maxArea=Math.max(maxArea,Math.min(height[l],height[r])*(r-l));

            if(height[l]<height[r]){
                l++;
            }
            else{
                r--;
            }
           
        }
        return maxArea;
        
    }
}